### PR TITLE
DDF-5867 fix table export Visible

### PR DESCRIPTION
**Pull request by andrewzimmer**
#### What does this PR do?
Export Visible always exports the first page of results, this updates table export to use the correct starting index for visible
#### Who is reviewing it? 
@hayleynorton @cassandrabailey293 @zta6 @willwill96 

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@bdthomson

#### How should this be tested?
- build / install / run
- ingest data and search for it
- export results via the table
- export Visible results on multiple different pages and verify correct results are present
- hide a result and verify it is not present when exporting visible results
- verify no other regression in exporting

#### What are the relevant tickets?
Fixes: #5867 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Review by zta6 — APPROVED** [review]

**Review by hayleynorton — APPROVED** [review]

File: ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java [hunk 1]
```diff
@@ -253,6 +373,64 @@ private void attachFileToResponse(
...
   }
 
+  private CollectionResultComparator getResultComparators(List<CqlRequest.Sort> sorts) {
+    CollectionResultComparator resultComparator = new CollectionResultComparator();
+    for (CqlRequest.Sort sort : sorts) {
+      Comparator<Result> comparator;
+
+      String sortType = sort.getAttribute();
+      SortOrder sortOrder =
+          (sort.getDirection() != null && sort.getDirection().equals("descending"))
```
Comment (bdeining) [inline ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java]: `SortOrder.DESCENDING`

File: ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java [hunk 2]
```diff
@@ -157,7 +238,48 @@ public Object handle(Request request, Response response) throws Exception {
...
+                .filter(
+                    result ->
+                        !cqlTransformRequest
+                            .getHiddenResults()
+                            .contains(result.getMetacard().getId()))
+                .collect(Collectors.toList());
+
+    QueryResponse combinedResponse =
+        new QueryResponseImpl(
+            new QueryRequestImpl(new QueryImpl(ECQL.toFilter(cqlRequests.get(0).getCql()))),
```
Comment (bdeining) [inline ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java:275]: is it possible for `cqlRequests` to be empty?

Comment (andrewzimmer) [reply ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java:275]: No, I have a check for that here https://github.com/codice/ddf-ui/pull/105/files#diff-4ff7d3773f55434d06f165c2a670d3edR215

**Review by bdeining — APPROVED** [review]

File: ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java [hunk 3]
```diff
@@ -253,6 +373,64 @@ private void attachFileToResponse(
...
+
+    AttributeType.AttributeFormat format =
+        result
+            .getMetacard()
+            .getMetacardType()
+            .getAttributeDescriptor(attributeName)
+            .getType()
+            .getAttributeFormat();
+
+    switch (format.toString()) {
```
Comment (mojogitoverhere) [inline ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java]: ❓ Any reason not to `switch` on the enum directly?

File: ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java [hunk 4]
```diff
@@ -118,22 +133,88 @@ public void setArguments(Map<String, Object> arguments) {
...
+
+    public void setSorts(List<CqlRequest.Sort> sorts) {
+      this.sorts = sorts;
+    }
+
+    public List<CqlRequest.Sort> getSorts() {
+      return this.sorts;
+    }
+
+    public List<CqlRequestImpl> getCqlRequests() {
```
Comment (mojogitoverhere) [inline ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java]: ✏️ This method seems out of place. What do you think about doing this transformation inline and removing this method?

File: ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java [hunk 5]
```diff
@@ -118,22 +133,88 @@ public void setArguments(Map<String, Object> arguments) {
...
+    private int count;
+    private List<CqlRequest.Sort> sorts;
+    private List<String> hiddenResults;
+
+    public CqlTransformRequest() {
+      this.searches = Collections.emptyList();
+      this.count = 0;
+      this.sorts = Collections.emptyList();
+      this.hiddenResults = Collections.emptyList();
+    }
```
Comment (mojogitoverhere) [inline ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java]: ✏️ You can initialize these directly and remove this constructor to save on a few lines of code
```
    private List<CqlRequestImpl> searches = Collections.emptyList();
    private int count = 0;
    private List<CqlRequest.Sort> sorts = Collections.emptyList();
    private List<String> hiddenResults = Collections.emptyList();
```

File: ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java [hunk 6]
```diff
@@ -157,7 +238,48 @@ public Object handle(Request request, Response response) throws Exception {
...
-    CqlQueryResponse cqlQueryResponse = cqlQueryUtil.executeCqlQuery(cqlRequest);
+    List<Result> results =
+        cqlRequests
+            .stream()
+            .map(
+                cqlRequest -> {
+                  CqlQueryResponse cqlQueryResponse = null;
+                  try {
+                    cqlQueryResponse = cqlQueryUtil.executeCqlQuery(cqlRequest);
+                  } catch (Exception e) {
```
Comment (mojogitoverhere) [inline ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java]: :question: Should we catch a more specific exception?

File: ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java [hunk 7]
```diff
@@ -253,6 +373,64 @@ private void attachFileToResponse(
...
+        result
+            .getMetacard()
+            .getMetacardType()
+            .getAttributeDescriptor(attributeName)
+            .getType()
+            .getAttributeFormat();
+
+    switch (format.toString()) {
+      case "STRING":
+        return attribute.getValue().toString().toLowerCase();
```
Comment (mojogitoverhere) [inline ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java]: ❗️ Can `attribute.getValue()` return null?

File: ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java [hunk 8]
```diff
@@ -118,22 +133,88 @@ public void setArguments(Map<String, Object> arguments) {
...
-    if (cqlRequest.getCql() == null) {
+    List<CqlRequest> cqlRequests =
+        cqlTransformRequest
+            .getCqlRequests()
+            .stream()
+            .filter(
+                cqlRequest ->
+                    cqlRequest.getCql() != null
+                        && (cqlRequest.getSrc() != null
+                            || CollectionUtils.isNotEmpty(cqlRequest.getSrcs())))
```
Comment (Lambeaux) [inline ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java:200]: ❓ Does `isNotEmpty` already check for `null`?

**Review by Lambeaux — APPROVED** [review]
Lots of @mojogitoverhere's comments are good. Approved assuming they are fixed.

**Review by mojogitoverhere — APPROVED** [review]

File: ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java (hunk 8 above)
Comment (andrewzimmer) [reply ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java:200]: It does, I'm checking the list of `srcs` in this line or for a single source in the line above

**Pull request merged**